### PR TITLE
Added CustomEvent "cookiebarConsent" to improve developer integration

### PR DESCRIPTION
**Pull request by Geolim4**
#129 Added CustomEvent "cookiebarConsent" to improve developer integration

### CODE REVIEW

**Comment by ToX82** [comment]
Awesome! Can you suggest a few lines of documentation so that I can explain the functionality on the website as well? A few examples would be great too.

I'll merge it as soon as I can though, as soon as I get back to the office in a few days :)

**Comment by Geolim4** [comment]
So basically it's simple:

When a developer want to integrate the cookie-bar, he may want to be notified of what the user choose about it's cookie consent. 
Does it refused the consent ? 
Did he gave it ? 
Or did he customized it's consent ?

The `cookiebarConsent` [CustomEvent](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent) aim to make the integration easier and avoid developer to make dirty trick using "onclick the read cookie" to get aware of the cookie consent.

So if a developer want to integrate the cookie-bar all he have to do is this code:

```ts
document.addEventListener('cookiebarConsent', (e) => (
     console.log(e.detail.consent);
));
```

The code above allows to asynchronously subscribe to the `cookiebarConsent` [CustomEvent](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent) and retrieve the user choice using `e.detail.consent`.

Basically the `e.detail.consent` is an `enum` of 3 possibles string values:

```ts
enum CookiebarConsent {
  CookieAllowed,
  CookieDisallowed,
  CookieCustomized,
}
```

This way you can avoid dirty code trick like I had to do:
https://github.com/Geolim4/In-Memoriam/blob/80b4817cfec0881046566a243a04cab2ac0ba7a8/src/ts/appCore.ts#L228
#

**Comment by Geolim4** [comment]
@ToX82 👋🏻

**Comment by ToX82** [comment]
Merged! Thanks @Geolim4 !

**Comment by Geolim4** [comment]
Hello @ToX82 

That's great, thank !

May I suggest you some improvements to your repository ?

1)
It would be more practical to put your website URL on the top of your repository but to also setup some keywords and a more detailed description, here's an example how I configured mine:

image

2) You may would like to create an issue template to help user fill better detailed issues ?
See [my repository](https://github.com/PHPSocialNetwork/phpfastcache/issues/new/choose) for an example that can be configured via the [.github](https://github.com/PHPSocialNetwork/phpfastcache/tree/master/.github/ISSUE_TEMPLATE) directory.

3)
You may want to setup a JS Linter (like eslint) ? This would make the code more "standardized" across your user's contributions :)

4)
Did you stopped making releases (last was sep 2021) ?
https://github.com/ToX82/cookie-bar/releases

Cheers,
Georges

**Comment by Geolim4** [comment]
@ToX82 ⬆️  😃

**Comment by ToX82** [comment]
Thanks, I've added some tags and the website, and a .eslintrc file. It was very needed indeed :)

As for the releases, I actually did stop making those as I don't feel the need for releases for this kind of packages. Most people use it from jsdelivr or you can just download the master branch (which is aligned with jsdelivr btw)

**Comment by Geolim4** [comment]
Great ! This tiny project has some true potential, you better fully embrace it ! 😃

**Comment by Geolim4** [comment]
@ToX82 I just implemented the JS events of this PR and it's working fine 😀 

➡️  https://in.memoriam.forces-francaises.fr/
